Invalid memory address or nil pointer dereference in s3_gate tests

On neofs-s3-gw v0.27.0 in the neofs-node autotests, the s3_gate tests fails:

```
RuntimeError: Failed to init s3 credentials because of error
Could not parse info from output
Enter password for /home/runner/work/neofs-node/neofs-node/neofs-testcases/TemporaryDir/5137298a-b153-41cb-9942-99edd7f9b6ab.json > 
2023-06-02T10:13:46.996Z	DEBUG	s3-authmate/main.go:688	prepare connection pool
2023-06-02T10:13:47.007Z	INFO	authmate/authmate.go:237	check container or create	{"cid": "11111111111111111111111111111111", "friendly_name": "47b73db7-def0-4b4e-933e-ca28ef6e165c", "placement_policy": "REP 2 IN X CBF 3 SELECT 2 FROM * AS X"}
2023-06-02T10:13:48.026Z	INFO	authmate/authmate.go:245	store bearer token into NeoFS	{"owner_tkn": "NRdrkLYdxg9fofWaeVRcfRhDT6Ta6NhJ73"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xca1ad3]

goroutine 1 [running]:
github.com/nspcc-dev/neofs-s3-gw/internal/neofs.(*NeoFS).CreateObject(0xc0000109c0, {0x12aba98, _}, {{0x0, 0x0}, {0x6c, 0xb4, 0x90, 0x31, 0x3a, ...}, ...})
	github.com/nspcc-dev/neofs-s3-gw/internal/neofs/neofs.go:278 +0x6d3
github.com/nspcc-dev/neofs-s3-gw/internal/neofs.(*AuthmateNeoFS).CreateObject(0xc0000120d8, {0x12aba98, 0xc0000a38c0}, {{{0xc000049700, 0x19, 0x19}}, {0x6c, 0xb4, 0x90, 0x31, ...}, ...})
	github.com/nspcc-dev/neofs-s3-gw/internal/neofs/neofs.go:538 +0x272
github.com/nspcc-dev/neofs-s3-gw/creds/tokens.(*cred).Put(0xc0000d7700, {0x12aba98, 0xc0000a38c0}, {0x6c, 0xb4, 0x90, 0x31, 0x3a, 0xe3, 0xf6, ...}, ...)
	github.com/nspcc-dev/neofs-s3-gw/creds/tokens/credentials.go:131 +0x2e7
github.com/nspcc-dev/neofs-s3-gw/authmate.(*Agent).IssueSecret(0xc0000d7b00, {0x12aba98, 0xc0000a38c0}, {0x12a2e60, 0xc000120008}, 0xc0000d7b50)
	github.com/nspcc-dev/neofs-s3-gw/authmate/authmate.go:250 +0xa65
main.issueSecret.func1(0xc000398400?)
	github.com/nspcc-dev/neofs-s3-gw/cmd/s3-authmate/main.go:379 +0xc9b
github.com/urfave/cli/v2.(*Command).Run(0xc0003d7440, 0xc000427e80)
	github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x5c8
github.com/urfave/cli/v2.(*App).RunContext(0xc00043c9c0, {0x12aba60?, 0xc000114000}, {0xc000128000, 0xf, 0xf})
	github.com/urfave/cli/v2@v2.3.0/app.go:313 +0xb36
github.com/urfave/cli/v2.(*App).Run(...)
	github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
	github.com/nspcc-dev/neofs-s3-gw/cmd/s3-authmate/main.go:147 +0x425
```

See the full report:
https://nspcc-dev.github.io/neofs-node/156/

Environment:
```
neo-go version
NeoGo
Version: 0.101.1
GoVersion: go1.19.7
==========================================
neofs-s3-authmate version
NeoFS S3 Authmate
Version: v0.27.0
GoVersion: go1.20.4
==========================================
NeoFS S3 Gateway
Version: v0.27.0
GoVersion: go1.20.4
==========================================
Check neofs-adm version
NeoFS Adm
Version: 60f10958 
GoVersion: go1.20.4
==========================================
neofs-ir version
NeoFS Inner Ring node
Version: 60f10958 
GoVersion: go1.20.4
==========================================
neofs-lens version
NeoFS Lens
Version: 60f10958 
GoVersion: go1.20.4
==========================================
neofs-cli version
NeoFS CLI
Version: 60f10958 
GoVersion: go1.20.4
```

--- a/internal/neofs/neofs.go
+++ b/internal/neofs/neofs.go
@@ -262,7 +262,7 @@ func (x *NeoFS) CreateObject(ctx context.Context, prm layer.PrmObjectCreate) (oi
 
 	if prm.BearerToken != nil {
 		prmPut.UseBearer(*prm.BearerToken)
-	} else {
+	} else if prm.PrivateKey != nil {
 		prmPut.UseSigner(neofsecdsa.SignerRFC6979(*prm.PrivateKey))
 	}
 
@@ -301,7 +301,7 @@ func (x *NeoFS) ReadObject(ctx context.Context, prm layer.PrmObjectRead) (*layer
 
 	if prm.BearerToken != nil {
 		prmGet.UseBearer(*prm.BearerToken)
-	} else {
+	} else if prm.PrivateKey != nil {
 		prmGet.UseSigner(neofsecdsa.SignerRFC6979(*prm.PrivateKey))
 	}
 
@@ -334,7 +334,7 @@ func (x *NeoFS) ReadObject(ctx context.Context, prm layer.PrmObjectRead) (*layer
 
 		if prm.BearerToken != nil {
 			prmHead.UseBearer(*prm.BearerToken)
-		} else {
+		} else if prm.PrivateKey != nil {
 			prmHead.UseSigner(neofsecdsa.SignerRFC6979(*prm.PrivateKey))
 		}
 
@@ -369,7 +369,7 @@ func (x *NeoFS) ReadObject(ctx context.Context, prm layer.PrmObjectRead) (*layer
 
 	if prm.BearerToken != nil {
 		prmRange.UseBearer(*prm.BearerToken)
-	} else {
+	} else if prm.PrivateKey != nil {
 		prmRange.UseSigner(neofsecdsa.SignerRFC6979(*prm.PrivateKey))
 	}
 
@@ -393,7 +393,7 @@ func (x *NeoFS) DeleteObject(ctx context.Context, prm layer.PrmObjectDelete) err
 
 	if prm.BearerToken != nil {
 		prmDelete.UseBearer(*prm.BearerToken)
-	} else {
+	} else if prm.PrivateKey != nil {
 		prmDelete.UseSigner(neofsecdsa.SignerRFC6979(*prm.PrivateKey))
 	}
 

